frontend/src/deployment/searchIndex.ts

```diff
@@ -17,121 +18,151 @@ export interface IndexedRuleWithDescription extends IndexedRule {
   descriptions?: Array<string>;
 }
 
-/**
- * Create the index store. This store is indexed by lunr and later used by the frontend.
- * Whenever the lunr index finds something it returns IDs. The frontend will look at
- * metadata corresponding to this ID in the index store.
- * @param rulesPath Path to the directory containing aggregated metadata and rules
- *                  descriptions in HTML format.
- */
-export function buildIndexStore(rulesPath: string):[Record<string,IndexedRuleWithDescription>, IndexAggregates] {
-  let ruleDirs = fs.readdirSync(rulesPath).filter((fileName) => {
-    const fullpath = path.join(rulesPath, fileName);
-    return fs.lstatSync(fullpath).isDirectory();
-  });
-  let allTags: { [id: string]: number } = {};
-  let allLangs: { [id: string]: number } = {};
-  let allQualityProfiles: { [id: string]: number } = {};
-  const indexedRecords = ruleDirs.map<[string, IndexedRuleWithDescription] | null>((ruleDir) => {
-    const allLanguages = fs.readdirSync(path.join(rulesPath, ruleDir))
-                            .filter((fileName) => fileName.endsWith('-metadata.json'))
-                            .map((fileName) => fileName.split('-')[0]);
-
-    let types = new Set<string>();
-    let severities = new Set<Severity>();
-    const all_keys = new Set<string>([ruleDir]);
-    const titles = new Set<string>();
-    const tags = new Set<string>();
-    const qualityProfiles = new Set<string>();
-    const descriptions = new Set<string>();
-    let prUrl = undefined;
-
-    allLanguages.forEach((lang) => {
-      // extract every word of every description of this rule
-      const descriptionPath = path.join(rulesPath, ruleDir, `${lang}-description.html`);
-      const descriptionStr = fs.readFileSync(descriptionPath).toString();
-      // Remove HTML tags from the description, extract unique words and normalize them.
-      // This reduces a bit the footprint of descriptions in the index.
-      const descriptionWords = stripHtml(descriptionStr).result.split(DESCRIPTION_SPLIT_REGEX)
-      descriptionWords.forEach((word) => descriptions.add(word));
-
-      // merge metadata fields of every version of this rule in a single indexed record
-      const metadataPath = path.join(rulesPath, ruleDir, `${lang}-metadata.json`);
-      const metadataStr = fs.readFileSync(metadataPath).toString();
-      const metadata = JSON.parse(metadataStr);
-
-      if (metadata.prUrl) {
-        prUrl = metadata.prUrl;
+function buildOneRuleRecord(allLanguages: string[],
+  rulesPath: string,
+  ruleDir: string) {
```

Comment:
```suggestion
function buildOneRuleRecord(allLanguages: string[], rulesPath: string, ruleDir: string) {
```
Feel free to discard this stylistic edit. Personally I find it more readable to have arguments on one line when they fit.